src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderRepository.cs

```diff
@@ -149,18 +150,24 @@ public Task PutAsync(SortedDictionary<int, ProvenBlockHeader> headers, HashHeigh
             {
                 this.logger.LogTrace("({0}.Count():{1})", nameof(headers), headers.Count());
 
-                using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
-                {
-                    transaction.SynchronizeTables(BlockHashHeightTable, ProvenBlockHeaderTable);
+                RetryStrategy.Run(
+                    DBreezeRetryOptions.Default,
+                    () =>
+                    {
+                        using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
+                        {
+                            transaction.SynchronizeTables(BlockHashHeightTable, ProvenBlockHeaderTable);
```

Comment:
Are these the only tables being modified? Would adding `Common` help resolve the underlying issue?

Reply:
What issue? the exception?

Reply:
![image](https://user-images.githubusercontent.com/29645989/52850786-7cb35c80-3168-11e9-8fba-05463e8c807c.png)

Reply:
Looks like NOT specifying all the tables may lead to intermittent deadlocks. Since this appears to be an intermittent issue maybe this would help?